Zend/zend_object_handlers.c

```diff
@@ -669,6 +674,16 @@ zval *zend_std_read_property(zval *object, zval *member, int type, void **cache_
 				retval = &EG(uninitialized_zval);
 			}
 			zval_ptr_dtor(&tmp_object);
+
+			if (UNEXPECTED(ZEND_CLASS_HAS_TYPE_HINTS(zobj->ce) &&
+				(prop_info = zend_object_fetch_property_type_info(Z_OBJCE_P(object), Z_STR_P(member), cache_slot)))) {
+				zval tmp, *val;
+
+				val = zend_verify_property_type(prop_info, retval, &tmp, (zobj->ce->__get->common.fn_flags & ZEND_ACC_STRICT_TYPES) != 0);
```

Comment:
Shouldn't this assign the result to `retval` rather than temporary? Otherwise we might return the wrong type if strict_types=0 and a conversion happens.